### DISCUSSION TITLE
Is inpaint in the Control module behaving as intended?

### DISCUSSION THREAD

**Discussion by Silanda** [vladmandic/sdnext#2856] [Q&A] [answered]
Okay, so my workflow using the img2img module goes like this: I upscale the original image using the SD Upscale script, and then I enhance detail by masking and inpainting certain areas (faces, hands, etc).

I can't inpaint a full 4K image at anything like a reasonable speed, so I set the inpaint resolution to something lower but still larger than the area to be inpainted, and set the inpaint area to masked. The area around the mask is then rendered with more detail and composited back into the full image.

I can't seem to replicate this workflow in the Control module. Rather than only rendering the inpainted area, it seems to render the entire image at the resolution set in the "Before" option, and outputs the final image at that resolution. "Input masked only" doesn't seem to have any effect, and nothing's being composited. In fact, messing with the mask settings with "input masked only" turned on seems to result in an image similar to what's being seen in #2846, rendering the unpainted mask over the resized image. No errors though.

Setting the "After" resolution to the original 4K image's resolution just seems to produce an output that's downscaled the original image, inpainted the whole image at the "before" resolution, and then upscaled it back.

This is on a current dev build and without using any controlnet functionality. 

I was just hoping for some input into whether this is user error on my part, or if the "input masked only" option is malfunctioning, before I open an issue.

**Comment by vladmandic** [comment, marked as answer]
its probably latter, i'm transferring this to an issue

**Comment by vladmandic** [comment]
should be fixed in latest dev.
